lib/search.rb

```diff
@@ -456,6 +456,19 @@ def self.advanced_filters
       )", tags)
   end
 
+  advanced_filter(/filetypes?:([a-zA-Z0-9,\-_]+)/) do |posts, match|
+    file_extensions = match.split(",")
+
+    posts.where("posts.id IN (
+      SELECT post_id FROM topic_links
+      WHERE extension IN (?)
+      UNION
+      SELECT post_uploads.post_id FROM uploads
+      JOIN post_uploads ON post_uploads.upload_id = uploads.id
+      WHERE lower(uploads.extension) IN (?)
+      )", file_extensions, file_extensions)
```

Comment:
You can use `:file_extensions` in the SQL and just pass a single option `{ file_extensions: file_extensions }`